Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM debian:bullseye-slim as luarocks-installer
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y --no-install-recommends \
+      build-essential \
+      libpcre2-dev \
+      libssl-dev \
+      libxml2 \
+      lua5.1 \
+      luarocks && \
+    luarocks install --tree /luarocks/usr/local luaossl && \
+    luarocks install --tree /luarocks/usr/local lrexlib-pcre2 && \
+    luarocks install --tree /luarocks/usr/local luautf8 && \
+    luarocks install --tree /luarocks/usr/local xmlua && \
+    luarocks install --tree /luarocks/usr/local base64 && \
+    # Here we do a bit of black magic to find which .so files are depended by the newly installed lua libraries:
+    # Start by finding all .so files in the install prefix, which is /luarocks/usr/local
+    find /luarocks/usr/local -type f -name '*.so' \
+      # Pipe the found .so through ldd, which will output all the system libraries used by the lua module
+      | xargs ldd  \
+      # Run through an awk filter, which will skip all standard .so files which are already in the production image
+      | awk '$3 && $1 !~ /libc\.so|vdso\.so|ld-linux|libpthread\.so|libm\.so|libdl\.so/ {print $3}' \
+      # Add those libraries to tar archive that will be sent to stdout. Note the -h flag, which is used to deference symlinks
+      | tar cf - -C / -h --verbatim-files-from -T - \
+      # Untar in the /luarocks directory, which is what we will unpack in the final image
+      | tar xf - -C /luarocks && \
+    # Same as above, but use dpkg -L to retrieve libraries not visible in ldd output. These are loaded through luajit FFI by xmlua
```

Comment:
Lol super fun, so in production dockerfile we'll have to consolidate this down to the single approach I think that is already used but fine for now.

Reply:
Yea, unfortunately xmlua is an exception because it relies on LuaJIT FFI to load libxml. Usually native Lua modules have an explicit link to system dependencies which would be visible in the `ldd` output.

In any case, I also think it is better to explicitly list dependencies, but I would like some extra time to write more Lua examples which will certainly require more dependencies. After we have a good set of Lua examples doing various tasks, we can define a standard library that would be included in the official image.